src/controllers/async-controller.ts

```diff
@@ -0,0 +1,20 @@
+import amqp from 'amqplib';
+
+const rabbitmqHost = process.env.RABBITMQ_HOST ?? "localhost";
+const rabbitmqUrl = `amqp://${rabbitmqHost}`;
+var connection: amqp.Connection;
+var channel: amqp.Channel;
+const queues: string[] = ['testQueue'];
+
+export async function initializeAsyncController() {
+    connection = await amqp.connect(rabbitmqUrl);
+    channel = await connection.createChannel();
+    queues.forEach(async (queue) => { await channel.assertQueue(queue); });
+
+    channel.consume('testQueue', (message) => { console.log(message?.content.toString()); });
+}
+
+export async function requestTest(message: string) {
+    channel.sendToQueue('testQueue', Buffer.from(message));
```

Comment:
Do we need to add channel.ack() to either of these functions? Does the message get removed from the queue automatically?

Reply:
Huh. Good question. The docs for amqplib say it defaults to manual so I should need one. However, in practice everything works fine. I'll push up a change that explicitly sets it to auto so I don't need the ack.

Future services may require an ack but the test point doesn't.